src/snowflake/snowpark/functions.py

```diff
@@ -7047,6 +7048,10 @@ def udf(
     external_access_integrations: Optional[List[str]] = None,
     secrets: Optional[Dict[str, str]] = None,
     immutable: bool = False,
+    native_app_params: Optional[
+        Dict[str, Any]
+    ] = None,  # This could look like native_app_params =
+    # {"schema": "some_schema", "application_roles": ["app_public", "app_admin"], "packages": ["a", "b"], "imports": ["c", "d"]}
```

Comment:
as discussed, I don't see that packages and imports would need to be added here. Schema and app roles are fair game though, depending on Snowpark's own plans for generation.